liberty-managed/src/main/java/org/jboss/arquillian/container/was/wlp_managed_8_5/WLPManagedContainer.java

```diff
@@ -913,7 +913,11 @@ private void checkApplicationStatus(Map<ObjectName, AppStatus> appMBeans, boolea
                if(targetState == true) {
                   String applicationState = (String)mbsc.getAttribute(appMBean, "State");
                   if(applicationState.contentEquals("STARTED")) {
-                     status = AppStatus.FINISHED;
+                      status = AppStatus.FINISHED;
+                  } else if (applicationState.contentEquals("INSTALLED")) {
+                      // If the application went from STARTING state to INSTALLED state, there might've been
+                      // a deployment issue. Check the logs if a deployment exception occurred.
+                      throwWrappedExceptionIfFoundInLog(appMBean.getKeyProperty("name"));
```

Comment:
We should throw a generic exception if we didn't find the exception in the log, otherwise we will still fail with timeout which doesn't make sense.